pkg/pillar/cmd/downloader/dirs.go

```diff
@@ -43,38 +54,32 @@ func clearInProgressDownloadDirs(ctx *downloaderContext) {
 			existingTargets[obj.Target] = true
 		}
 	}
-
-	// Now remove the in-progress dirs
-	workingDirTypes := []string{getPendingDir()}
-
-	for _, dirName := range workingDirTypes {
-		if _, err := os.Stat(dirName); err == nil {
-			err := filepath.Walk(dirName, func(walkPath string, fi os.FileInfo, err error) error {
```

Comment:
This old code does a walk which descends into subdirs. I think all we have is a flat directory now, but @petr-zededa can you please double check that we do not need to descend into subdirs.

Reply:
Yes, we have flat directory with files for non-oci downloads inside (blob itself and progress file).